src/apikey/lib/redis.js

```diff
@@ -11,29 +12,40 @@ if (!fs.pathExists('../apikey.json')) {
   logger.warn('No API key are set');
 }
 
-if (process.env.NODE_ENV === 'production') {
-  apiKeys = require('../apikey.json');
-} else {
-  apiKeys = require('../apikey-dev.json');
-}
-
 const redisClient = redis.createClient({
   host: config.get('redis.host'),
   port: config.get('redis.port'),
   password: config.get('redis.password'),
 });
 
+redisClient.get = util.promisify(redisClient.get);
+redisClient.del = util.promisify(redisClient.del);
 redisClient.ping = util.promisify(redisClient.ping);
+redisClient.set = util.promisify(redisClient.set);
+redisClient.keys = util.promisify(redisClient.keys);
+redisClient.flushall = util.promisify(redisClient.flushall);
 
 redisClient.on('error', (err) => {
   logger.error(`Error in redis ${err}`);
 });
 
-const apiKeysJSON = Object.keys(apiKeys);
+const load = async () => {
+  if (process.env.NODE_ENV === 'production') {
+    apiKeys = await fs.readFile(path.resolve(__dirname, '..', 'apikey.json'));
+  } else {
+    apiKeys = await fs.readFile(path.resolve(__dirname, '..', 'apikey-dev.json'));
+  }
+  apiKeys = JSON.parse(apiKeys);
 
-const load = () => {
-  apiKeysJSON.forEach((key) => {
-    redisClient.set(key, `${JSON.stringify(apiKeys[key])}`, redis.print);
+  const apiKeysJSON = Object.keys(apiKeys);
+
+  apiKeysJSON.forEach(async (key) => {
+    try {
+      await redisClient.set(key, `${JSON.stringify(apiKeys[key])}`);
+    } catch (err) {
+      logger.error(`Cannot load ${key} with ${JSON.stringify(apiKeys[key])} on redis`);
+      logger.error(err);
+    }
   });
```

Comment:
```suggestion
  await Promise.all(
    Object.entries(apiKeys).map(async ([keyId, keyValue]) => {
      try {
        await redisClient.set(keyId, `${JSON.stringify(keyValue)}`);
      } catch (err) {
        logger.error(`Cannot load ${keyId} with ${JSON.stringify(keyValue)} on redis`);
        logger.error(err);
      }
    })
  );
```
`forEach()` will not wait for the promises to be resolved, so if you want to be sure that the keys are loaded when the `load()` function resolves, you should use either `Promise.all` (to run them in parallel) or a `for loop` (to run them sequentially).